addons/web_editor/static/src/scss/web_editor.frontend.scss

```diff
@@ -55,6 +55,19 @@
     > * {
         padding: var(--grid-item-padding-y) var(--grid-item-padding-x) !important;
     }
+
+    // Adapt the horizontal margins of the first inner row of a grid item, to
+    // make them compensate the grid item horizontal padding (to avoid an
+    // overflow).
+    .o_grid_item .row {
+        --grid-inner-row-gutter-x: clamp(0px, 2 * var(--grid-item-padding-x), 30px);
+        margin-left: calc(-0.5 * var(--grid-inner-row-gutter-x));
+        margin-right: calc(-0.5 * var(--grid-inner-row-gutter-x));
+
+        .row {
+            --grid-inner-row-gutter-x: var(--gutter-x);
+        }
```

Comment:
Is this because we can now have a grid inside a grid ? Do you have an exemple of such a case that I could test ? 🙏🏻

Reply:
It is to compensate the margins of the first row only :slightly_smiling_face: 

=> In 16.0, the only occurrence of the bug was with the "Team" snippet, which has a row as direct child of a grid item. But in 17.0, the bug also happen with any grid item having an inner row (so not necessarily a direct child like with the "Team" snippet) in mobile view :grimacing: 
An example is is with the form on the "Contact Us" page (as explained in the commit message): as you can see, the row of the form is not the direct child, so I had to remove `>` from the rule :+1:  However, this is not the only row: each form field is also a row with columns 
This means that the `.o_grid_item .row` rule would also apply on them (because this is "a row inside a grid item"), which we don't want :+1: 
So in order to only compensate the margin of the first row, I had to add this rule for the rows inside this row so they use their original gutter-x: "a row inside a grid item must compensate the margins, but the rows inside that row don't need to" (there is no CSS selector to say "the first occurrence of..." :smiling_face_with_tear: )

Note that there can never be a grid inside a grid (I blocked it) :wink:

Reply:
Okay, makes sense, sorry if you had to repeat what was in the commit message, I better understand now 😄

Reply:
No problem :wink: